enhancements/installer/openstack-ipi-failure-domains.md

```diff
@@ -0,0 +1,340 @@
+
+When deploying nodes on separate subnets, the user will need to update the `machineNetwork` option to list all the possible CIDRs on which the machines can be:
+
+```yaml
+networking:
+  ...
+  machineNetwork:
+  - cidr: 192.168.25.0/24
+  - cidr: 192.168.26.0/24
+  - cidr: 192.168.27.0/24
+  - ...
+```
+
+ The `machineNetwork` is the source for creating the security groups rules during installation. There's a [known bug][bz-2095323] in the OpenStack implementation that only takes into account the first CIDR.
+
+For now, `machinesSubnet` is required to be set when deploying nodes on separate Failure Domains, and is then used to identify where to create the API and Ingress ports.
+
+The OpenShift administrator will be able to adjust the number of computes nodes for a specific Failure Domain, but only on day 2.
+
+Also, the installer will provide some validations in order to avoid deployment errors:
+
+- A maximum of 3 `failureDomains` can be provided to the control plane machine pool. OpenShift only supports 3 control plane nodes for now.
+- Having `failureDomains` in machinepool without `machinesSubnet` is an error.
+- machinepool `zones` for both compute and storage can't be used together with machinepool `failureDomains`.
+- Check that `subnets` in `failureDomains` actually exist (the same validation as machinesSubnet). Note that we can't easily verify if the subnet can actually be used for a given availability zone. This is up to the OpenShift administrators to figure out which subnet they can use for which domain.
+- Check that `failureDomain` subnets have a matching CIDR in `machineNetwork`.
+
+### Risks and Mitigations
+
+- UX will be reviewed by Field Engineers, partnering with a customer who needs this feature. Also our QE will help to review it.
+
+### Drawbacks
+
+- The networking resources are pre-created and managed by the OpenStack administrators, and therefore not managed by the OpenShift cluster. Changes in the networking infrastructure will have to be planned and accordingly updated in the cluster.
+  This is not a new requirement by the way, this comes from the fact it relies on the BYON (bring-your-own-network) feature. If a network goes on maintenance, the machines connected to that network will have to be migrated or redeployed else where.
+- Adding new parameters (especially when they contain sub-parameters) is never easy for the users but we'll make sure to choose easy names and that they're all documented.
+
+## Design Details
+
+### Open Questions
+
+- What maximum latency and bandwidth should we require between the failure domains?
+- Where should the failure domains be defined? Potential places would be the platform or the machine pool scope.
+- How do we specify where to create the API and Ingress VIPs ports? Is it OK to reuse machinesSubnet for this purpose? Isn't it confusing?
+- Should we discover the CIDR of the provided subnets and add them to MachineNetwork CIDR list automatically?
```

Comment:
Could we have a platform operator to do this?

@JoelSpeed ?

Reply:
It's better to be declarative than infer and discover, is a MachineNetwork an existing thing? Can the installer populate this information somewhere? 

I'm trying to make a push for more install time knowledge to be populated on the `Infrastructure.Spec.PlatformSpec.<platform>` so that operators have a source of truth for this information

Reply:
Agreed with @JoelSpeed that declarative is better. This is the current proposal.

Regarding the second question, I believe MachineNetwork not to be used outside of the installer. See relevant conversation
https://github.com/openshift/enhancements/pull/1167#discussion_r924108557

> I'm trying to make a push for more install time knowledge to be populated on the Infrastructure.Spec.PlatformSpec.<platform> so that operators have a source of truth for this information

I'm also in favor of this. It's a great thing to explore, but out of scope for this enhancement I think.